### ISSUE TITLE
Testing output limited to max 32 characters

### ISSUE THREAD

**Issue by lynoure** [rust-cli/book#13]
The way shown for testing output is not viable when the output is longer than 32. There is a way to change the test slightly to allow testing with longer inputs. That way the readers of this book will not have to go on googling spree to it out elsewhere, like I had to.

I'll be probably making a pull request on this later this week.

**Comment by killercup** [comment]
Sorry, just to clarify, how is it limited to 32? Haven't looked at this
book in a while and don't remember encountering any such limitations.

Looking forward to any clarification PRs :)

**Comment by lynoure** [comment]
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f547ae748163d0b4f55b21488bbaee52 illustrates what I'm talking about. 

I'll get to the PR later :)

**Comment by mig4** [comment]
Now I'm intrigued ;) The example you posted appears to work OK. What was the issue you were seeing @lynoure?

**Comment by lynoure** [comment]
TL;DR: Maybe it was just beginner confusion. Or maybe [it was fixed in rust 1.47.0](https://turreta.com/2020/11/14/rust-array-size-limit-to-32-no-more-in-1-47-0-release/). Anyway, _feel free to close_.

Alas, it's now a year later and I've forgotten pretty much everything. 
So, let's do some archeology: I don't have much except what I can see from the Discord #beginners logs and I see from my code that I ended up asserting in a totally different way in my code as the result of whatever that was. 

My own notes in my journal on this are not extensive enough to make much sense from in hindsight either:
```
Following the instructions from https://rust-cli.github.io/book/tutorial/testing.html for testing output is interesting:

It only works if equality is one way around, such as assert_eq!(result, expected), the other way not, when assert_eq!(expected, result) .

The whole thing fails altogether when the output is longer, with "error[E0277]: arrays only have std trait implementations for lengths 0..=32", so I need to try with the advice I got:

"So thats a reference to an array, which currently have a weird limitation length for some APIs - you can avoid the issue by using slices, either via &[a, b, ...][..], or indireclty by using &vec![a, b, ...] to get a reference to a vector (which can auto-deref to a slice)"
```

**Comment by mig4** [comment]
Oh right, an interesting bit of history there ;) Thank you for sharing the details.

Anyway, you're right it looks like one way or another something was fixed and this is no longer happening so the issue can be closed.
